Teams_App/src/Components/SearchBarComponent/SearchBarComponent.js

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+import "./SearchBarComponent.css";
+import CARDS_LIST from "../../Data/TeamData";
+
+const SearchBarComponent = ({ setCardsList }) => {
+  /**
+   * getFilteredData
+   * @param {*} search
+   * @returns List of filtered cards matching the searched pattern
+   */
+  const getFilteredData = (search) => {
+    return CARDS_LIST.filter((card) =>
+      card.name.toLowerCase().includes(search.toLowerCase())
+    );
+  };
+
+  //   State variables for type time and search string
+  const [searchText, setSearchText] = useState("");
+  const [typeTime, setTypeTime] = useState(true);
+
+  /**
+   * inputChangeHandler
+   * @param {*} event
+   * Event handler for the form -> onInput and onSubmit
+   */
+  const inputChangeHandler = (event) => {
+    event.preventDefault();
+    if (event._reactName == "onInput") {
+      if (typeTime == true) {
+        setCardsList(getFilteredData(event.target.value));
+        setTypeTime(false);
```

Comment:
Never try to access the state in a component like that. We are never sure which state we get. Don't access them, use them for conditional rendering etc. Also if you read docs, you'll know that setState if run two times doesn't mean that state updates twice. Refer beta docs. Instead, make a local variable and try debouncing with it.